app/src/store/cart.ts

```diff
@@ -95,6 +96,102 @@ export default function useCartStore() {
     }
   }
 
+  // convert the amount to a different token (direction allows for moving to/from the tokenAddress)
+  function getConvertedAmount(amount: number, tokenAddress: string, direction = 1) {
+    const exchangeRate = quotes.value[tokenAddress] ?? 0;
+
+    return amount * (direction == 1 ? exchangeRate : 1 / exchangeRate);
+  }
+
+  // Linear Interpolation function to find the point on the curve that amount corresponds to
+  function lerp(x_lower: number, x_upper: number, y_lower: number, y_upper: number, x: number) {
+    return y_lower + ((y_upper - y_lower) * (x - x_lower)) / (x_upper - x_lower);
+  }
+
+  // returns the prediction curve for this grant in the given round
+  function getPredictionsForGrantInRound(grantId: string, round: GrantRound) {
+    const data = ((grantRoundsCLRData && grantRoundsCLRData.value && grantRoundsCLRData.value) || {}) as {
+      [grantRound: string]: GrantRoundCLR;
+    };
+    const roundData = data[round.address] || {};
+
+    return roundData.predictions && roundData.predictions[Number(grantId)];
+  }
+
+  // given the grantId and the contribution amount find all matching in all rounds
+  function getPredictionForAmount(grantId: string, amount: number, token: TokenInfo) {
```

Comment:
Lots of these methods don't feel like cart-specific methods, and feel more like dcurve methods or utils. They also don't rely on the cart state, so I think they should be pulled out to live somewhere else (partly because the cart file is getting pretty big too)

Reply:
I've abstracted some of the less relevant ones over to `dcurve`, but kept anything that depends on `quotes`